### PR TITLE
Fix bad code in c plus and python

### PR DESCRIPTION
**Pull request by guochaorong**
I am wondering if the script python/paddle/fluid/layers/metric.py  have been used  so far.  the  ‘auc’ method in it seems does not work.

### CODE REVIEW

**Comment by guochaorong** [comment]
resolve issues：
https://github.com/PaddlePaddle/Paddle/issues/11257
https://github.com/PaddlePaddle/Paddle/issues/11256

File: paddle/fluid/platform/device_tracer.cc [hunk 1]
```diff
@@ -130,8 +130,6 @@ void CUPTIAPI bufferRequested(uint8_t **buffer, size_t *size,
   uint8_t *buf = reinterpret_cast<uint8_t *>(malloc(kBufSize + kAlignSize));
   *size = kBufSize;
   *buffer = ALIGN_BUFFER(buf, kAlignSize);
-  free(buf);
```
Comment (guochaorong) [inline paddle/fluid/platform/device_tracer.cc]: I didn't notice that it was requesting a buffer.... and I found that has been freed through the cuptiActivityRegisterCallbacks ， and finally freed at the end of bufferCompleted function.

**Comment by panyx0718** [comment]
nice! We can discuss about this tool later.

**Comment by guochaorong** [comment]
Thank you.  It's my pleasure to discuss it with you

**Pull request merged**